distributed/deploy/local.py

```diff
@@ -94,6 +94,7 @@ class LocalCluster(SpecCluster):
 
     def __init__(
         self,
+        name=None,
```

Comment:
The position of this keyword is breaking some tests which specify `n_workers` by positions (e.g. `distributed/deploy/tests/test_local.py::test_scale_up_and_down`)

```
    async with LocalCluster(
        0,
        scheduler_port=0,
        processes=False,
        silence_logs=False,
        dashboard_address=None,
        asynchronous=True,
    ) as cluster:
```

Perhaps we should start switching some classes to use keyword-only arguments in their constructors to avoid this in the future?

Reply:
Thanks @jrbourbeau. I think that would be sensible.

I'm happy to go through and update the tests to use explicit kwargs. Perhaps one day when our minimum Python version is 3.8 we could use the PEP570 syntax and disable positional args on many classes altogether.

Reply:
Others may be using `LocalCluster(n_workers)` in the wild.  This may cause some user frustration. 

Thoughts on moving the `name=` parameter to later in this list of parameters?